src/Shouldly/ShouldlyExtensionMethods/ShouldBeEnumerableTestExtensions.cs

```diff
@@ -154,6 +199,22 @@ private static List<T> GetDuplicates<T>(IEnumerable<T> items)
             return duplicates;
         }
 
+        // Could possible remove the above implementation and pass an ObjectComparer like the generic ShouldBe methods do.
+        // Also added generics instead of object as this is probably how it should have been implemented.
+        private static List<T> GetDuplicates<T>(IEnumerable<T> items, IEqualityComparer<T> comparer)
+        {
+            var list = new List<T>();
+            var duplicates = new List<T>();
+
+            foreach (var o1 in items)
+            {
+                duplicates.AddRange(list.Where(o2 => o1 != null && comparer.Equals(o1, o2)));
```

Comment:
This means `null` is never detected as a duplicate.

Reply:
Also, NUnit has struggled with performance issues by not sticking with hash lookups for these comparisons. I'd recommend using HashSet or Dictionary to make use of `IEqualityComparer<T>.GetHashCode`.

Reply:
https://github.com/shouldly/shouldly/pull/701